git_machete/annotation.py

```diff
@@ -0,0 +1,59 @@
+import re
+from typing import Callable, Optional
+
+from git_machete import utils
+
+
+class Qualifiers:
+    rebase: Optional[bool]
+    merge: Optional[bool]
+    push: Optional[bool]
+
+    def __init__(self, annotation: str):
+        self._annotation = annotation
+        self._annotation_without_qualifiers = annotation
+        self._rebase_text = ''
+        self._push_text = ''
+        self.rebase = None
+        self.push = None
+
+        match_pattern: Callable[[str], str] = lambda text: f'.*\\b{text}=no\\b.*'
+        sub_pattern: Callable[[str], str] = lambda text: f'[ ]?{text}=no[ ]?'
+
+        rebase_match = re.match(match_pattern('rebase'), annotation)
+        if rebase_match:
+            self.rebase = False
+            self._rebase_text = 'rebase=no'
+            self._annotation_without_qualifiers = re.sub(sub_pattern('rebase'), ' ', self._annotation_without_qualifiers)
+
+        push_match = re.match(match_pattern('push'), annotation)
+        if push_match:
+            self.push = False
+            self._push_text = 'push=no'
+            self._annotation_without_qualifiers = re.sub(sub_pattern('push'), ' ', self._annotation_without_qualifiers)
+
+    def get_annotation_text_without_qualifiers(self) -> str:
+        return self._annotation_without_qualifiers.strip()
+
+    def get_qualifiers_text(self) -> str:
+        return f'{self._rebase_text.strip()} {self._push_text.strip()}'.replace('  ', ' ').strip()
+
+
+class Annotation:
+    text: str
+    text_without_qualifiers: str
+    qualifiers_text: str
+    qualifiers: Qualifiers
+
+    def __init__(self, text: str):
+        self.text = text
+        self.qualifiers = Qualifiers(text)
+        self.text_without_qualifiers = self.qualifiers.get_annotation_text_without_qualifiers()
+        self.qualifiers_text = self.qualifiers.get_qualifiers_text()
+
+    def get_color_formatted_text(self):
+        annotation_text = f"  {utils.dim(self.text_without_qualifiers)}"
+        if self.qualifiers_text != '':
+            annotation_text += ' ' if self.text_without_qualifiers != '' else ''
+            annotation_text += utils.colored(s=self.qualifiers_text, color=utils.AnsiEscapeCodes.BLUE)
```

Comment:
It's pretty dark on default iTerm2 theme:

<img width="850" alt="image" src="https://user-images.githubusercontent.com/3383210/205082632-7778d8e7-1b03-4c85-bb7b-8ee263b03cd0.png">

In fact... maybe instead of introducing yet another color, just **underline** the qualifier parts of the annotation?